packages/dev/inspector/src/components/actionTabs/tabs/tools/metadataEditorComponent.tsx

```diff
@@ -0,0 +1,206 @@
+
+import "./metadataEditor.scss";
+
+let currentEntity: any;
+const initialWindowSize = { width: 512, height: 512 };
+interface IMetadataEditorComponentProps {
+    scene: Scene;
+    globalState: GlobalState;
+}
+export const MetadataEditorComponent: React.FC<IMetadataEditorComponentProps> = (props: IMetadataEditorComponentProps) => {
+    const selectionUpdated = (entity: any) => {
+        currentEntity = entity;
+    };
+
+    const observableSelection: Observable<any> = props.globalState.onSelectionChangedObservable;
+    const observerMetaPopup: Nullable<Observer<any>> = observableSelection.add(selectionUpdated, -1, false, null, false);
+
+    const startMetadataEditorPopup = (scene: Scene): void => {
+        Inspector._CreatePersistentPopup(
+            {
+                props: {
+                    id: "metadata-editor",
+                    title: "Metadata Editor",
+                    onClose: () => observableSelection.remove(observerMetaPopup),
+                    size: initialWindowSize,
+                },
+                children: <MetadataEditorPopupComponent scene={scene} globalState={props.globalState} />,
+            },
+            document.body
+        );
+    };
+    return <ButtonLineComponent label="Open Metadata Editor" onClick={() => startMetadataEditorPopup(props.scene)} />;
+};
+
+interface IMetadataEditorPopupComponentProps {
+    scene: Scene;
+    globalState: GlobalState;
+}
+
+enum MetaDataTypes {
+    UNDEFINED = "undefined",
+    NULL = "null",
+    STRING = "string",
+    JSON = "JSON",
+}
+const MetadataEditorPopupComponent: React.FC<IMetadataEditorPopupComponentProps> = (props: IMetadataEditorPopupComponentProps) => {
+    const [currentEntityUniqueId, setCurrentEntityUniqueId] = useState<string | null>(null);
+    const [currentEntityId, setCurrentEntityId] = useState("");
+    const [currentEntityName, setCurrentEntityName] = useState("");
+    const [currentEntityMetadata, setCurrentEntityMetadata] = useState<string>("");
+    const [metadataPropType, setMetadataPropType] = useState<string>(MetaDataTypes.UNDEFINED);
+    const [isValidJson, setIsValidJson] = useState(false);
+    const [prettyJson, setPrettyJson] = useState(true);
+    const [jsonParseErrorMsg, setJsonParseErrorMsg] = useState<string | null>(null);
+
+    const stringifyEntityType = () => {
+        if (Object.prototype.hasOwnProperty.call(currentEntity, "metadata")) {
+            const meta = currentEntity.metadata;
+            if (isString(meta)) return MetaDataTypes.STRING;
+            return meta === null ? MetaDataTypes.NULL : MetaDataTypes.JSON;
+        }
+        return MetaDataTypes.UNDEFINED;
+    };
+
+    const isString = (input: any): boolean => {
+        return typeof input === "string" || input instanceof String;
+    };
+    const parsableJson = (object: Object): boolean => {
+        if (!object) return false;
+        return !!JSON.parse(JSON.stringify(object));

```

Comment:
This could throw an exception if there is an issue with the object.

Reply:
Done! Please review other commits, further improvements